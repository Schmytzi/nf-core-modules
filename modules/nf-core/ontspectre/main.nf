process ONTSPECTRE {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/73/73e5ebe79fbffbe451a05bf9182220a70f46ed3187891d8ab9033119f4919dc8/data':
        'community.wave.seqera.io/library/ont-spectre:0.3.2--adfae189059be3d9' }"

    input:
    tuple val(meta), path(summary), path(regions_bed), path(vcf), path(tbi)
    tuple val(meta2), path(fasta)
    tuple val(meta3), path(fai)
    tuple val(meta4), path(metadata)  // reference metadata, optional
    tuple val(meta5), path(blacklist) // optional
    val bin_size


    output:
    tuple val(meta), path("out/*.bed")                  , emit: bed
    tuple val(meta), path("out/*.vcf.gz")               , emit: vcf
    tuple val(meta), path("out/*.tbi")                  , emit: tbi
    tuple val(meta), path("out/*.spc.gz")               , emit: spc
    tuple val(meta), path("out/windows_stats/*.csv")    , emit: windows_stats
    tuple val(meta), path("out/predicted_karyotype.txt"), emit: karyotype
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def meta_arg = args.contains("--metadata") || !metadata ? "" : "--metadata $metadata"
    def blacklist_arg = args.contains("--blacklist") || !blacklist ? "" : "--blacklist $blacklist"
    """
    spectre \\
        CNVCaller \\
        $args \\
        $meta_arg \\
        $blacklist_arg \\
        --bin-size $bin_size \\
        --threads $task.cpus \\
        --sample-id $prefix \\
        --coverage . \\
        --snv $vcf \\
        --reference $fasta \\
        --output-dir out \\


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ont-spectre: \$(spectre version |& sed '1!d ; s/spectre::INFO> Spectre version: //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir out
    cd out
    mkdir windows_stats
    touch ${prefix}.bed ${prefix}.vcf.gz.tbi predicted_karyotype.txt windows_stats/${prefix}.csv
    echo "" | gzip > ${prefix}.vcf.gz
    echo "" | gzip > ${prefix}.spc.gz
    cd ..

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ont-spectre: \$(spectre version |& sed '1!d ; s/spectre::INFO> Spectre version: //')
    END_VERSIONS
    """
}
