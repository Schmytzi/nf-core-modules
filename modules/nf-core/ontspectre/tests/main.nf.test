nextflow_process {

    name "Test Process ONT_SPECTRE"
    script "../main.nf"
    process "ONTSPECTRE"

    tag "modules"
    tag "modules_nfcore"
    tag "ontspectre"

    test("homo sapiens - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file("summary"),
                    file("regions"),
                    file("vcf.gz"),
                    file("vcf.gz.tbi")
                ]
                input[1] = [
                    [ id: 'test' ],
                    file("fasta")
                ]
                input[2] = [
                    [ id: 'test' ],
                    file("fai")
                ]
                input[3] = [ [], [] ]
                input[4] = [ [], [] ]
                input[5] = 1000
                """

            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}