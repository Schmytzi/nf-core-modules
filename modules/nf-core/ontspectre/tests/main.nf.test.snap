{
    "homo sapiens - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spc.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "predicted_karyotype.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    "versions.yml:md5,836df2b55b02061bfe39fcf7c79995f2"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "karyotype": [
                    [
                        {
                            "id": "test"
                        },
                        "predicted_karyotype.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "spc": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spc.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "tbi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,836df2b55b02061bfe39fcf7c79995f2"
                ],
                "windows_stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-03T08:59:11.844190948"
    }
}